lib/devices/phone_device_manager.js

```diff
@@ -0,0 +1,78 @@
+import assert from 'assert';
+import Debug from 'debug';
+
+import Testdroid from 'testdroid-client';
+
+let debug = Debug('taskcluster-docker-worker:devices:phoneManager');
+
+export default class PhoneDeviceManager {
+  constructor(config) {
+    assert(config.deviceManagement.phone.sims, "Must supply the number of sims required");
+    assert(config.deviceManagement.phone.type, "Must supply the type of phone");
+    assert(config.testdroid, "Must supply testdroid configuration");
+    assert(config.testdroid.url, "Must supply testdroid cloud url");
+    assert(config.testdroid.username, "Must supply testdroid cloud username");
+    assert(config.testdroid.password, "Must supply testdroid cloud password");
+
+    this.config = config;
+    this.client = new Testdroid(
+        config.testdroid.url,
+        config.testdroid.username,
+        config.testdroid.password
+    );
+    this.deviceFilter = {
+      'type': config.deviceManagement.phone.type,
+      'sims': config.deviceManagement.phone.sims
+    };
+  }
+
+  async devices() {
+    let devices = await this.client.getDevices(this.deviceFilter);
+    let deviceList = devices.map((device) => {
+      return new Phone(device);
+    });
+
+    debug(`List of ${deviceList.length} phones created`)
+    return deviceList;
+  }
```

Comment:
So won't all workers have a same list of devices?

I feel like it's bad idea to recreate the `Phone` objects... maybe it would be smarter to maintain a  map from `phone.id` to phone object... and then add phone devices that shows up, and remove phone devices that disappears from the API...

I argue for this mostly because having an object representing a phone isn't super fun, if there is two instances of the object representing the same phone device. And `getAvailableDevice` is probably used by some consumer that keeps the `Phone` object around until the task is done.